spicy/toolchain/src/compiler/codegen/grammar-builder.cc

```diff
@@ -127,16 +127,24 @@ struct Visitor : public hilti::visitor::PreOrder<Production, Visitor> {
                 // Skipping not supported
             }
 
+            else if ( const auto& size = AttributeSet::find(n.attributes(), "&size") )
+                // FIXME(bbannier): the referenced node seems to go away the node in `skip` ends up pointing to junk.
```

Comment:
Any idea why this is or what I could do here to make sure the ref stays valid @rsmmr?

Reply:
The issue here was how conditional fields were handled. This was already broken for conditional fields with literal values. I added a patch & test which fixes that.